### PR TITLE
(maint) make apt testing more stable, cleanup

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

**Comment by tphoney** [comment]
This seems to fix most of our jenkins / ci issues, as well as clean up the code and tests.
There is an outstanding issue of debian 8 in travis CI failing, this only happened yesterday, proven by #765 
I will tackle that issue separately

File: manifests/source.pp [hunk 1]
```diff
@@ -29,17 +28,9 @@
     $_release = $release
   }
 
-  # Some releases do not support https transport with default installation
```
Comment (ekohl) [inline manifests/source.pp]: Removal of this is a regression. It's a feature we depend on.

Comment (tphoney) [reply manifests/source.pp]: https://tickets.puppetlabs.com/browse/MODULES-7540 has been created, this was my fault. I have a fix incoming, and will start a release soon after.

Comment (ekohl) [reply manifests/source.pp]: Thanks, just ran into it but hadn't had time to create an issue myself. Luckily for me I noticed it on a CI job and not in production.

**Pull request merged**